woocommerce/woocommerce-ios
# [Improvement in Order Edit/Details] Show state name instead of code for Japan and Turkey

Closes: #5851 


## Description

With this PR we align with https://github.com/woocommerce/woocommerce-android/pull/5603 and show the state name instead of code for addresses in Japan and Turkey. These countries have non- (easily) identifiable state codes, such as JP01 or TR07. 
To fix this, we query our database to retrieve these states' names, and if we don't have them stored, we show the code as before. This approach has the advantage of not requiring an extra backend call, with the drawback of failing if we didn't request the list of countries before. 
Most likely the user had to select a country and state with the app when creating the order, thus requesting and storing the country's information. If the user created the order through a different platform, e.g. web or the app on another device, we won't show the name. In case this becomes an issue we could download the countries information on the app startup when opening the order. That being said, Android's approach is the same and there were not many complaints so far.


## Testing instructions


1. Go to Orders
2. Tap on + to create a new order
3. Add the customers' information. Select Japan and Hokkaido.
4. Go back to the order editing screen. Check the address. Before it showed JP01, now Hokkaido.
5. Go to the order details. Check that the address is also correct.

Try also with a Turkish address.


## Screenshots


### Before

<img src="https://github.com/woocommerce/woocommerce-ios/assets/1864060/c605c8d1-cad3-4e3b-afe2-5477f6aa7739" width="375">
<img src="https://github.com/woocommerce/woocommerce-ios/assets/1864060/7fc6ba61-e0c3-459b-9404-722f01b9c742" width="375">

### After
<img src="https://github.com/woocommerce/woocommerce-ios/assets/1864060/510b7ecc-76f2-4af9-bbb6-de256373cedc)" width="375">
<img src="https://github.com/woocommerce/woocommerce-ios/assets/1864060/1d224878-5db0-468d-80c5-b8746a5ef61f" width="375">

---
